vermouth/molecule.py

```diff
@@ -96,7 +94,7 @@ def match(self, node, key):
         """
         Apply the comparison.
         """
-        return node.get(key) in self.value
+        return node.get(key) in self.value or super().match(node, key)
```

Comment:
My idea was that two Predicates that are the same (same type, same value) should match with eachother. It's probably better to give the baseclass an `__eq__` method, and compare for equality.

Reply:
My idea is that `Choice(['GLY', 'ALA', 'VAL'])` is indeed equal to itself, but it does not match it. But my idea is valid because I think about the final matching against the molecule. You think about a matching against an intermediate structure. They should probably be 2 different methods.